[ADD] Camera Page

LfrPaul opened the pull request:
J'ai ajouté 2 routes vu que y'a le cas où on prend la première photo, et le cas où on prend une autre photo (donc le texte en haut de page changeait)

ElementState et CurrentElementViewModel -> Pour l'instant y'a que le name puisque j'avais besoin que de ça

StepState -> Pour le champ Photo, j'ai fait une Map<Url,Photo>, comme ça on pourra facilement retrouver l'élément lorsqu'on voudra le supprimer vu que la fonction de suppression prend en param une Url

J'ai modifié légérement le composant PhotoDescription pour mettre Uri au lieu de URI (sinon j'arrivais pas à afficher l'image) et aussi ajouter la fonctionnalité de clear l'input en appuyant sur la croix

Le Modifier.coloredShadow c'est un truc que j'ai trouvé sur internet pour avoir une bonne ombre sur le texte en haut de la page (sinon le rendu était vraiment moche)

Rendu : 
![Screenshot_20240517_181238_Immobylette](https://github.com/Immobylette/immobylette--app-mobile/assets/78427088/a02d90da-0b9c-4fad-b027-9f581f2805ab)

A l'appui sur le bouton photo : 
![Screenshot_20240517_181314_Immobylette](https://github.com/Immobylette/immobylette--app-mobile/assets/78427088/c6f84dc1-02d5-4c7c-987f-bb3746c2b5e4)

Commit:
[UPDATE] CameraPage
Ajout de la modal

Commit:
[UPDATE] CurrentElementViewModel
Remove default value

LfrPaul commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Page prise de photo](https://trello.com/c/WYd1Q4Zi/82-page-prise-de-photo)

patacoing commented on app/src/main/java/com/immobylette/appmobile/step/StepState.kt:
```diff
@@ -0,0 +1,11 @@
+data class StepState (
+    val description: String = "",
+    val errorDescription: String? = "",
+    val state: String = "", // TODO mettre le type enum
```
manque le type ElementState

patacoing commented on app/src/main/java/com/immobylette/appmobile/element/camera/CameraPage.kt:
```diff
@@ -0,0 +1,237 @@
+@OptIn(ExperimentalPermissionsApi::class, ExperimentalMaterial3Api::class)
+@Composable
+fun CameraPage(
+    takingAnotherPicture: Boolean,
```
mettre une valeur par défaut à false comme ça dans la navigation pour l'une des routes, pas besoin de passer le param

patacoing commented on app/src/main/java/com/immobylette/appmobile/element/camera/CameraPage.kt:
```diff
@@ -0,0 +1,237 @@
+                        }
+                    }
+                )
+                com.immobylette.appmobile.ui.shared.component.Button (
```
importer com.immobylette.appmobile.ui.shared.component.Button pour ne pas à avoir à mettre tout le chemin ici

patacoing commented on app/src/main/java/com/immobylette/appmobile/element/camera/CameraPage.kt:
```diff
@@ -0,0 +1,237 @@
+    }
+    if(displayModal) {
+        AlertDialog(
+            onDismissRequest = {}
```
je ne sais pas si la callback onDismissRequest s'eécute quand on clique en dehors de la modale. Mais si c'est le cas, il faudrait mettre displayModal à false pour que l'utilisateur puisse reprendre une photo

Commit:
rebase

Commit:
[UPDATE] onDismissRequest to hide modal
Import Button in CameraPage
Default value for takingAnotherPicture in CameraPage
ElementState in StepState

patacoing committed:
fix/IMB-82

* [ADD] CameraPage
Pas encore de modal pour la description

* [UPDATE] CameraPage
Ajout de la modal

* [UPDATE] CurrentElementViewModel
Remove default value

* [FIX] photo is taking entire width

* [FIX] photo is taking entire width

---------

Co-authored-by: Paul <paullefevre59190@gmail.com>
